src/testExplorer/testExplorer.ts

```diff
@@ -131,38 +156,34 @@ export function watchTestFiles(controller: vscode.TestController) {
 		globalState.fileWatcher?.dispose();
 	}
 	globalState.fileWatcher = vscode.workspace.createFileSystemWatcher("**/*.{yaml,yml}");
+	
+	let debounceTimeout: NodeJS.Timeout | undefined;
+	
+	const debouncedRefresh = () => {
+		if (debounceTimeout) {
+			clearTimeout(debounceTimeout);
+		}
+		debounceTimeout = setTimeout(() => {
+			refreshTestExplorer(controller);
+		}, 300);
+	};
+
 	globalState.fileWatcher.onDidCreate(uri => {
-		addTestFile(controller, uri);
-		globalState.treeDataProvider?.refresh();
+		debouncedRefresh();
 	});
+	
 	globalState.fileWatcher.onDidChange(uri => {
-		updateTestFile(controller, uri);
-		globalState.treeDataProvider?.refresh();
+		debouncedRefresh();
```

Comment:
Do we need a slightly longer debounce on editing? Concerned about a lot of IO intensive work for a short pause in typing.

Reply:
Not sure i follow. 🤔